src/Registry.php

```diff
@@ -449,6 +454,8 @@ public function offsetUnset($offset)
      * @return  mixed  The value of the that has been set.
      *
      * @since   1.0.0
+     *
+     * @deprecated The $separator parameter will be removed in version 4.
```

Comment:
Hmmm, does this not just set the separator for the current path given to the set method and then its dropped? So I am not sure if this needs to be changed actually it is a very useful feature actually since you can with a path add the separator for that moment to be '/', and yet later when you call the data it falls back to the class separator. They are by design exclusive. Well at least from what I can see.

Reply:
Also, the empty check below (L462) should be changed to `$separator !== null`

Reply:
the usage of the separator per function call is inconsistent (if I checked it correctly it only exists for the set function).

For consistent reasons I would limit it to a write only property in the constructor.
if you really need it for your "set" operation you can still use str_replace('/', '.', $textSeparetedBySlash).

the alternative is to add the separator to each function.

Reply:
You are probably right that a simple string replace could be used instead, although it may be more tedious. I typically prefer not to fix things that are not broken. The dynamic separators are used in the set functions to split the path, but in the get function, they fall back to the global/class separator. Therefore, I do not see any inconsistencies here. The set functions use the dynamic separator (local or global), while the get functions use the class property separator.

As someone who knows the CMS better than I do, do you know if the registry anywhere in the CMS uses this dynamic separator function? I could run a search, but I would like to know how much impact removing this feature would have on the CMS, since it appears to be a feature by design.

Reply:
tbh I don't know if the cms uses a dynamic separator. but the inconsistency for example in the append (not really sure what it does anyway, it uses a hardcoded separator) and remove function

Reply:
> it is a very useful feature actually since you can with a path add the separator for that moment to be '/'

It can be migrated, old:
```php
$a = new Registry(['foo' => 'bar']);
$data = ['the/path' => 'test'];
$a->set('foo2', $data, '/');
```

New:
```php
$a = new Registry(['foo' => 'bar']);
$b = new Registry(['the/path' => 'test'], '/');
$a->set('foo2', $b->toArray());
```

Having a "clean" set/get will allow us to have a RegistryInterface in future, at least in theory :wink:

Reply:
Okay so be it, but let it be noted that I was very happy to not change this behavior.